src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ObjectModelConverters.cs

```diff
@@ -238,6 +286,45 @@ internal static void FixUpTestCase(this TestCase testCase, string? testAssemblyP
         testCase.ExecutorUri = new(Constants.ExecutorUri);
     }
 
+    private static bool TryGetMethodIdentifierProperty(TestCase testCase, [NotNullWhen(true)] out TestMethodIdentifierProperty? methodIdentifierProperty)
+    {
+        string? managedType = testCase.GetPropertyValue<string>(ManagedTypeProperty, defaultValue: null);
+        string? managedMethod = testCase.GetPropertyValue<string>(ManagedMethodProperty, defaultValue: null);
+        // NOTE: ManagedMethod, in case of MSTest, will have the parameter types.
+        // So, we prefer using it to display the parameter types in Test Explorer.
+        if (RoslynString.IsNullOrEmpty(managedType) || RoslynString.IsNullOrEmpty(managedMethod))
+        {
+            methodIdentifierProperty = null;
+            return false;
+        }
+
+        methodIdentifierProperty = GetMethodIdentifierPropertyFromManagedTypeAndManagedMethod(managedType, managedMethod);
+        return true;
+    }
+
+    private static TestMethodIdentifierProperty GetMethodIdentifierPropertyFromManagedTypeAndManagedMethod(
+        string managedType,
+        string managedMethod)
+    {
+        ManagedNameParser.ParseManagedMethodName(managedMethod, out string methodName, out int arity, out string[]? parameterTypes);
+        if (arity != 0)
+        {
+            methodName = $"{methodName}`{arity.ToString(CultureInfo.InvariantCulture)}";
+        }
+
+        parameterTypes ??= [];
+
+        ManagedNameParser.ParseManagedTypeName(managedType, out string @namespace, out string typeName);
+
+        // In the context of the VSTestBridge where we only have access to VSTest object model, we cannot determine ReturnTypeFullName.
+        // For now, we lose this bit of information.
+        // If really needed in the future, we can introduce a VSTest property to hold this info.
+        // But the eventual goal should be to stop using the VSTestBridge altogether.
+        // TODO: For AssemblyFullName, can we use Assembly.GetEntryAssembly().FullName?
+        // Or alternatively, does VSTest object model expose the assembly full name somewhere?
```

Comment:
@nohwnd can you comment here pls?

Reply:
We don't have assembly full name in the data, the test cases don't have to be produced from the current process or source.  

If we want to populate that data, it might be better to ask for the actual type in the namespace, and grab the assembly information from there. Because then we don't assume the entry assembly is non-null, and we don't assume it is the same as the source of the test, we just assume that the type is available, and can decide to fail if it is not, or not fail and do something else.